src/evaluate/visualization.py

```diff
@@ -0,0 +1,224 @@
+
+    def _scale_data(self, data, ranges):
+        """Scales data[1:] to ranges[0]"""
+        for d, (y1, y2) in zip(data[1:], ranges[1:]):
+            assert (y1 <= d <= y2) or (y2 <= d <= y1)
+        x1, x2 = ranges[0]
+        d = data[0]
+        sdata = [d]
+        for d, (y1, y2) in zip(data[1:], ranges[1:]):
+            sdata.append((d - y1) / (y2 - y1) * (x2 - x1) + x1)
+        return sdata
+
+    def plot(self, data, *args, **kwargs):
+        """Plots a line"""
+        sdata = self._scale_data(data, self.ranges)
+        self.ax1.plot(self.angle, np.r_[sdata, sdata[0]], *args, **kwargs)
+        self.plot_counter = self.plot_counter + 1
+
+    def fill(self, data, *args, **kwargs):
+        """Plots an area"""
+        sdata = self._scale_data(data, self.ranges)
+        self.ax1.fill(self.angle, np.r_[sdata, sdata[0]], *args, **kwargs)
+
+    def use_legend(self, *args, **kwargs):
+        """Shows a legend"""
+        self.ax1.legend(*args, **kwargs)
+
+    def set_title(self, title, pad=25, **kwargs):
+        """Set a title"""
+        self.ax.set_title(title, pad=pad, **kwargs)
+
+
+def radar_plot(data, model_names, invert_range=[]):
+    """
+    `data`: list of `dict`s of metric + value pairs.
+        E.g. data = [{"accuracy": 0.9, "precision":0.8},{"accuracy": 0.7, "precision":0.6}]
+    `names`: list of `str`s with model names
+        E.g. names = ["model1", "model 2", ...]
+    `invert_range`: list of `str`s with the metrics to invert (in cases when lower is better, e.g. speed)
+        E.g. invert_range=["latency_in_seconds"]
+    """
+    if isinstance(data, list) is False:
+        raise ValueError("The input must be a list of dicts of metric + value pairs")
+    data = pd.DataFrame(data)
+    variables = data.keys()
+    if all(x in variables for x in invert_range) is False:
+        raise ValueError("All of the metrics in `invert_range` should be in the data provided.")
+    min_max_per_variable = data.describe().T[["min", "max"]]
+    min_max_per_variable["min"] = min_max_per_variable["min"] - 0.1 * (
+        min_max_per_variable["max"] - min_max_per_variable["min"]
+    )
+    min_max_per_variable["max"] = min_max_per_variable["max"] + 0.1 * (
+        min_max_per_variable["max"] - min_max_per_variable["min"]
+    )
+
+    ranges = list(min_max_per_variable.itertuples(index=False, name=None))
+    ranges = [
+        (max_value, min_value) if var in invert_range else (min_value, max_value)
+        for var, (min_value, max_value) in zip(variables, ranges)
+    ]
+    format_cfg = {
+        "rad_ln_args": {"visible": True},
+        "outer_ring": {"visible": True},
+        "angle_ln_args": {"visible": True},
+        "rgrid_tick_lbls_args": {"fontsize": 12},
+        "theta_tick_lbls": {"fontsize": 12},
+        "theta_tick_lbls_pad": 3,
+        "theta_tick_lbls_brk_lng_wrds": True,
+    }
+    fig = plt.figure(figsize=(6, 12), dpi=300)
```

Comment:
Similar to above `config` discussion. I think in the case of fig we shouldn't modify the systems default at all. We can add a `fig=None` keyword as well and then do the following:
```python
if fig is None:
    fig = plt.figure()
```

Reply:
otherwise people can specify a figure size?